### PR TITLE
ENH: Added Index param to to_dict like in to_json #46398

### LINKED ISSUE #46398: Add `index` param to `to_dict` like in `to_json`.
`df.to_json(orient='split', index=False)` is really nice. `df.to_dict(orient='split')` basically does the same thing, but I get a dictionary back instead of a string. For symmetry, I'd like `to_dict` to also have an `index` param, so I can keep from having to `del dictionary['index']` to make them really similar.

### PR DESCRIPTION
**Pull request by vruthik**
- [X] closes #46398 
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [X] Added an entry in the latest `doc/source/whatsnew/v1.5.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by vruthik** [comment]
Hello, it's my first time contributing! I'm not sure why pre-commit.ci is timing out. I ran pre-commit locally and all the checks seemed to pass fine. Any suggestions on what might be going wrong?

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1909,43 +1914,77 @@ def to_dict(self, orient: str = "dict", into=dict):
             elif orient.startswith("i"):
                 orient = "index"
 
+        if not index and orient not in ["split", "tight"]:
```
Comment (phofl) [inline pandas/core/frame.py:1917]: I am not sure this is desireable. The default config of index should not lead to a ValueError

File: pandas/tests/frame/methods/test_to_dict.py [hunk 2]
```diff
@@ -344,3 +344,19 @@ def test_to_dict_orient_tight(self, index, columns):
         roundtrip = DataFrame.from_dict(df.to_dict(orient="tight"), orient="tight")
 
         tm.assert_frame_equal(df, roundtrip)
+
+    def test_to_dict_index_orient_split(self):
```
Comment (phofl) [inline pandas/tests/frame/methods/test_to_dict.py:348]: Could you add the gh reference?

File: pandas/core/frame.py (hunk 1 above)
Comment (vruthik) [reply pandas/core/frame.py:1917]: Hey phofl! Thanks for the feedback! I assume you mean the default value for the index parameter - the default value for the index parameter is True, so it wouldn't lead to a ValueError. This check was directly taken from the implementation of the index parameter in 'to_json'. It checks that index is only True if orient is 'split' or 'tight'. A value error would be raised only when index is set to False AND orient is not 'split' or 'tight'.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in contributing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**